app/uk/gov/hmrc/gform/service/RepeatingComponentService.scala

```diff
@@ -152,20 +156,20 @@ class RepeatingComponentService @Inject() (val sessionCache: SessionCacheConnect
     } else {
       val groupFieldValue = repeatingGroupsFound.head
       val fieldsInGroup = cacheMap.getEntry[List[List[FieldValue]]](groupFieldValue.id.value).getOrElse(Nil).flatten
-      fieldsInGroup.size
+      Future.successful(fieldsInGroup.size)
     }
   }
 
-  private def getFormFieldIntValue(expr: TextExpression, data: Map[FieldId, Seq[String]]) = {
+  private def getFormFieldIntValue(expr: TextExpression, data: Map[FieldId, Seq[String]]): Future[Int] = {
```

Comment:
I don;t think this needs to return a future, I think it is better to wrap this function's result in a Future.successful where it is used